### DISCUSSION TITLE
Windows GPU support?

### DISCUSSION THREAD

**Discussion by wamppus** [winedarksea/AutoTS#164] [Q&A]
Is there a way to get gpu support working on windows systems? I noticed in the documentation that said it was only available that is was only useable on Linux.

**Comment by winedarksea** [comment]
GPU is used for only a few models: tensorflow (via sklearn -Regression models written by me), pytorch-forecasting (uses pytorch), and gluonts (uses mxnet) are the only ones that come to mind.

Installation then depends on each of those upstream packages: Tensorflow, MXNet, and Pytorch being setup to utilize GPU in your local environment, then AutoTS will utilize the GPU automatically. Go to the source documentation for each and they will guide on the GPU setup. Let me know if you run into issues.

I've found MXNet for GluonTS very well optimized for my Ryzen 5950X, so wouldn't bother with the GPU version as it isn't any faster on my RTX3060 12 GB. You might find what I wrote here helpful: https://syllepsis.live/2022/06/19/pytorch-tensorflow-and-mxnet-on-gpu-in-the-same-environment-and-gpu-vs-cpu-performance/ the TLDR for this is that because these neural networks are relatively small (compared to image processing and text monsters, we don't have that size of data here), and my CPU is a nice one, GPU's are often more trouble and cost to setup than they are worth.

**Reply by winedarksea** [reply]
Having totally ignored the Windows component of this, I should add I have gotten Tensorflow and Pytorch working just fine on Windows on GPU, I believe MXNet is the only problem for Windows.

**Comment by wamppus** [comment]
Thanks for your answer. It's awesome you're supporting this package to level. I'm currently running this on a laptop with an i7 10870h (laptop cpu) and a laptop 3080. I'm thinking my next step is to try and get the intel optimization packages installed and working?

do you still maintain the "pip install autots[additional]" function? When I originally wrote my code on collab I was able to install everything easily with that command and most models worked excellently, however, trying to set up the environment locally, i got an error using that command. something about a string error. I ended up following your installation guides but still getting some errors on models. i'll have to investigate these farther.  

the last question I did have is: after running a few csv files with model lists and parameters. is there a way i can combine them into my own "best model list" and only run those? i know you have a "run only one model" in the documentation, is the method similar?

**Reply by winedarksea** [reply]
sorry for the delayed reply.
Hmm, no, I don't test the additional installs option because I've found it doesn't work optimally on all systems, and can't with just one option: linux vs windows, arm vs x86... But I will check it and see.
Some errors are inevitable, I wouldn't worry too much about getting everything working as there is plenty of similarity between different models.

As for making your own best model list, you can do the import and export of templates, which I imagine you've seen in the production_example. The actual object results _object_ has a `concat` option for all results but that won't help with exported csvs.  What do I do in this case? I open them up in an Excel and manually combine. I have never written anything else because I am usually doing custom 'cherry picking' of results anyway which is best done in a spreadsheet tool.

If you do end up making a highly tailored template that you like and want to share, I can add it to the templates file. I've always had a (somewhat slowly maintained) general template, but it was built from the start with the possibility of storing other types of templates for specific use cases.
